cmake/compiler_support.cmake

```diff
@@ -0,0 +1,261 @@
+
+Find_package(Perl)
+
+if (NOT PERL_FOUND )
+	message(FATAL_ERROR "Perl not found")
+endif()
+
+set(MASM2GAS_PATH ${OMR_ROOT}/tools/compiler/scripts/masm2gas.pl CACHE INTERNAL "MASM2GAS PATH")
+
+# Given a prefix, and a list of arguments, prefix the list of arguments and
+# assign to out: ie, addprefix(out "-I" "a;b;c") should set out to
+# "-Ia;-Ib;-Ic".
+function(addprefix out prefix) 
+   set(ret "")
+   foreach(var IN ITEMS ${ARGN}) 
+      list(APPEND ret "${prefix}${var}")
+   endforeach()
+   set(${out} ${ret} PARENT_SCOPE)
+endfunction(addprefix) 
+
+
+# Platform setup code!
+# TODO: THis needs to be abstracted for cross platform builds
+
+set(TR_TARGET_ARCH    x     CACHE INTERNAL "The architecture directory used for the compiler code (x, p, or z)")
+set(TR_TARGET_SUBARCH amd64 CACHE INTERNAL "The subarchitecture directory used for the compiler code. May be empty (i386 or amd64)")
+set(CMAKE_ASM-ATT_FLAGS "--64 --defsym TR_HOST_X86=1 --defsym TR_HOST_64BIT=1 --defsym BITVECTOR_64BIT=1 --defsym LINUX=1 --defsym TR_TARGET_X86=1 --defsym TR_TARGET_64BIT=1" CACHE INTERNAL "ASM FLags")
+
+# Mark a target as consuming the compiler components. 
+# 
+# This supplements the compile definitions, options and 
+# include directories.
+# 
+# Call as make_compiler_component(<target> COMPILER <compiler name>)
+function(make_compiler_target TARGET_NAME) 
+   cmake_parse_arguments(TARGET 
+      "" # Optional Arguments
+      "COMPILER" # One value arguments
+      ""
+      ${ARGN}
+      )
+
+   set(COMPILER_DEFINES  ${${TARGET_COMPILER}_DEFINES})
+   set(COMPILER_ROOT     ${${TARGET_COMPILER}_ROOT})
+   set(COMPILER_INCLUDES ${${TARGET_COMPILER}_INCLUDES})
+   message("making ${TARGET_NAME} into a compiler target,for ${TARGET_COMPILER}.")
+   message("Defines are ${COMPILER_DEFINES}, arch is ${TR_TARGET_ARCH} and ${TR_TARGET_SUBARCH}") 
+   target_compile_features(${TARGET_NAME} PUBLIC cxx_static_assert)
+
+   target_include_directories(${TARGET_NAME} BEFORE PRIVATE 
+      ${COMPILER_INCLUDES}
+      )
+
+   # TODO: Extract into platform specific section.
+   target_compile_definitions(${TARGET_NAME} PRIVATE
+      BITVECTOR_BIT_NUMBERING_MSB
+      UT_DIRECT_TRACE_REGISTRATION
+      TR_HOST_64BIT
+      LINUX
+      TR_HOST_X86
+      TR_TARGET_X86
+      TR_TARGET_64BIT
+      ${COMPILER_DEFINES} 
+      )
+endfunction(make_compiler_target)
+
+# Filter through the provided list, and rewrite any 
+# .asm files to .s files, and add the .s file to the list 
+# in lieu of the .asm file. 
+#
+# Inspired by this post:  https://cmake.org/pipermail/cmake/2010-June/037733.html
+function(masm2gas_asm_files out_var compiler)
+
+   # This portion should be abstracted away, and the ATT_INCLUDES below should be made 
+   # into target properties to avoid polluting the rest of the info 
+   set(MASM2GAS_INCLUDES ${${compiler}_INCLUDES} $ENV{J9SRC}/oti)
+   addprefix(MASM2GAS_INCLUDES "-I" ${MASM2GAS_INCLUDES})
+   set(MASM2GAS_FLAGS --64                   )
```

Comment:
Mostly was there for alignment at one point -- but things changed so the alignment got messed up. I'll drop the spaces.